client/core/bond.go

```diff
@@ -150,337 +150,304 @@ func minBondLifetime(net dex.Network, bondExpiry int64) time.Duration {
 			} else {
-				// TODO: subject, detail := c.formatDetails(...)
-				details := fmt.Sprintf("Bond %v for %v refunded in %v, reclaiming %v of %v after tx fees",
-					bondIDStr, dc.acct.host, refundCoinStr, wallet.amtString(refundVal),
-					wallet.amtString(bond.Amount))
-				c.notify(newBondRefundNote(TopicBondRefunded, string(TopicBondRefunded),
-					details, db.Success))
-			}
-
-			err = c.db.BondRefunded(dc.acct.host, assetID, bond.CoinID)
-			if err != nil { // next DB load we'll retry, hit bondAlreadySpent, and store here again
-				c.log.Errorf("Failed to mark bond as refunded: %v", err)
+				refundCoinStr, refundVal = refundCoin.String(), refundCoin.Value()
 			}
-
-			spentBonds = append(spentBonds, &bondID{assetID, bond.CoinID})
-			transactedAssets[assetID] = struct{}{}
 		}
-
-		// Delete the now-spent bonds from the expiredBonds slice.
-		dc.acct.authMtx.Lock()
-		for _, spentBond := range spentBonds {
-			for i, bond := range dc.acct.expiredBonds {
-				if bond.AssetID == spentBond.assetID && bytes.Equal(bond.CoinID, spentBond.coinID) {
-					dc.acct.expiredBonds = cutBond(dc.acct.expiredBonds, i)
-					break // next spentBond
-				}
-			}
+		// RefundBond increases reserves when it spends the bond, adding to
+		// the wallet's balance (available or immature).
+
+		// If the user hasn't already manually refunded the bond, broadcast
+		// the refund txn. Mark it refunded and stop tracking regardless.
+		if bondAlreadySpent {
+			c.log.Warnf("Bond output not found, possibly already spent or never mined! "+
+				"Marking refunded. Backup refund transaction: %x", bond.RefundTx)
+		} else {
+			// TODO: subject, detail := c.formatDetails(...)
+			details := fmt.Sprintf("Bond %v for %v refunded in %v, reclaiming %v of %v after tx fees",
+				bondIDStr, dc.acct.host, refundCoinStr, wallet.amtString(refundVal),
+				wallet.amtString(bond.Amount))
+			c.notify(newBondRefundNote(TopicBondRefunded, string(TopicBondRefunded),
+				details, db.Success))
 		}
-		expiredStrength := sumBondStrengths(dc.acct.expiredBonds)
-		dc.acct.authMtx.Unlock()
 
-		for assetID := range transactedAssets {
-			c.updateAssetBalance(assetID)
+		err = c.db.BondRefunded(dc.acct.host, assetID, bond.CoinID)
+		if err != nil { // next DB load we'll retry, hit bondAlreadySpent, and store here again
+			c.log.Errorf("Failed to mark bond as refunded: %v", err)
 		}
 
-		bondAsset := bondAssets[bondAssetID]
-		if bondAsset == nil {
-			if targetTier > 0 {
-				c.log.Warnf("Bond asset %d not supported by DEX %v", bondAssetID, dc.acct.host)
+		spentBonds = append(spentBonds, &bondID{assetID, bond.CoinID})
+		assetIDs[assetID] = struct{}{}
+	}
+
+	return assetIDs, spentBonds, nil
+}
+
+// clearSpentBonds removes all bonds that have been successfully refunded from
+// a dexConnection's expiredBonds list and returns the strength of the remaining
+// bonds.
+func (c *Core) clearSpentBonds(dc *dexConnection, bondCfg *dexBondCfg, spentBonds []*bondID) int64 {
```

Comment:
Why does this one need to be separate, rather than part of `refundExpiredBonds`?

Reply:
No need, I've moved it.